common/script/libs/achievements.js

```diff
@@ -0,0 +1,322 @@
+  } else {
+    rebirthTitle = i18n.t('rebirthBegan', language);
+  }
+
+  if (!user.achievements.rebirthLevel) {
+    rebirthText = i18n.t('rebirthOrbNoLevel', language);
+  } else if (user.achievements.rebirthLevel < 100) {
+    rebirthText = i18n.t('rebirthOrb', {level: user.achievements.rebirthLevel}, language);
+  } else {
+    rebirthText = i18n.t('rebirthOrb100', language);
+  }
+
+  _add(result, {
+    key: 'rebirth',
+    title: rebirthTitle,
+    text: rebirthText,
+    icon: 'achievement-sun',
+    earned: Boolean(user.achievements.rebirths),
+  });
+
+  return result;
+}
+
+function _getSeasonalAchievements (user, language) {
+  let result = {};
+
+  _addPlural(result, user, {path: 'habiticaDays', language});
+  _addPlural(result, user, {path: 'habitBirthdays', language});
+
+  let spellAchievements = ['snowball', 'spookySparkles', 'shinySeed', 'seafoam'];
+  spellAchievements.forEach(path => {
+    _addSimpleWithCount(result, user, {path, language});
+  });
+
+  let questAchievements = ['dilatory', 'stressbeast', 'burnout', 'bewilder'];
+  questAchievements.forEach(path => {
+    _addQuest(result, user, {path, language});
+  });
+
+  _addPlural(result, user, {path: 'costumeContests', language});
+
+  let cardAchievements = ['greeting', 'thankyou', 'nye', 'valentine', 'birthday'];
+  cardAchievements.forEach(path => {
+    _addSimpleWithCount(result, user, {path, key: `${path}Cards`, language});
+  });
+
+  return result;
+}
+
+function _getSpecialAchievements (user, language) {
+  let result = {};
+
+  _addPlural(result, user, {path: 'habitSurveys', language});
+
+  let contribKey = 'contributor';
+  let contribContent = achievsContent[contribKey];
+  let contributorAchiev = {
+    key: contribKey,
+    text: i18n.t(contribContent.textKey, language),
+    icon: contribContent.icon,
+    earned: Boolean(user.contributor && user.contributor.level),
+  };
+  if (user.contributor && user.contributor.level) {
+    contributorAchiev.value = user.contributor.level;
+    contributorAchiev.title = contribText(user.contributor, user.backer, language);
+  } else {
+    contributorAchiev.value = 0;
+    contributorAchiev.title = i18n.t(contribContent.titleKey, language);
+  }
+  _add(result, contributorAchiev);
+
+  if (user.backer && user.backer.npc) {
+    _addSimpleWithCustomPath(result, user, {key: 'npc', path: 'backer.npc', language});
+  }
+
+  if (user.backer && user.backer.tier) {
+    _addSimpleWithCustomPath(result, user, {key: 'kickstarter', path: 'backer.tier', language});
+  }
+
+  if (user.achievements.veteran) {
+    _addSimple(result, user, {path: 'veteran', language});
+  }
+
+  if (user.achievements.originalUser) {
+    _addSimple(result, user, {path: 'originalUser', language});
+  }
+
+  return result;
+}
+
+// Build and return the given user's achievement data.
+achievs.getAchievementsForProfile = function getAchievementsForProfile (user, language) {
+  index = 0;
+  let result = {
```

Comment:
If the achievements in each category can be an array, then the `index` field isn't needed at all!